Modules/_ssl.c

```diff
@@ -984,6 +1020,10 @@ _ssl__SSLSocket_do_handshake_impl(PySSLSocket *self)
     /* XXX If SSL_do_handshake() returns 0, it's also a failure. */
     do {
         PySSL_BEGIN_ALLOW_THREADS
+        // XXX This isn't thread safe anymore, but locking this can cause a
```

Comment:
Perhaps add a note to the ssl module documentation about a threading restriction on do_handshake()?

Reply:
I don't think it's needed, nobody should be trying to call `do_handshake()` twice, let alone across multiple threads.